src/internal/m365/backup.go

```diff
@@ -186,3 +188,56 @@ func verifyBackupInputs(sels selectors.Selector, cachedIDs []string) error {
 
 	return nil
 }
+
+func (ctrl *Controller) CollectMetadata(
+	ctx context.Context,
+	r kinject.RestoreProducer,
+	man kopia.ManifestEntry,
+	errs *fault.Bus,
+) ([]data.RestoreCollection, error) {
+	var (
+		paths = []path.RestorePaths{}
+		err   error
+	)
+
+	for _, reason := range man.Reasons {
+		filePaths := [][]string{}
+
+		switch reason.Service() {
+		case path.GroupsService:
+			filePaths, err = groups.MetadataFiles(ctx, reason, r, man.ID, errs)
+			if err != nil {
+				return nil, err
+			}
+		default:
+			for _, fn := range graph.AllMetadataFileNames() {
+				filePaths = append(filePaths, []string{fn})
+			}
+		}
+
+		for _, fp := range filePaths {
+			rp, err := path.BuildRestorePaths(
+				reason.Tenant(),
+				reason.ProtectedResource(),
+				reason.Service(),
+				reason.Category(),
+				fp,
+			)
```

Comment:
We'll get the habit formed one of these days 😉 
```suggestion
				fp)
```

Reply:
I'm getting a linter for this. 🙈